components/autofill/src/sync/address/incoming.rs

```diff
@@ -68,7 +72,64 @@ impl ProcessIncomingRecordImpl for AddressesImpl {
         LEFT JOIN addresses_data l ON s.guid = l.guid
         LEFT JOIN addresses_tombstones t ON s.guid = t.guid";
 
-        common_fetch_incoming_record_states(tx, sql, |row| Ok(InternalAddress::from_row(row)?))
+        Ok(tx.query_rows_and_then_named(
```

Comment:
Oh it's sad that this logic has to exist in two places. But I'm imagining keeping this in `common.rs` would have become unruly. 😣

Reply:
Yeah, it's a bt unfortunate, but the duplication makes more sense once the encryption is taken into account.